### ISSUE TITLE
Running cloud cleaner on AWS errors on deleting security groups

### ISSUE THREAD

**Issue by jsdoeung-pivotal** [pivotal-gss/pcf-cloud-cleaner#5]
**Error below:** 

```
Processing Region [us-east-1]
    Terminating EC2 Instances...
{
    "TerminatingInstances": [
        {
            "CurrentState": {
                "Code": 32,
                "Name": "shutting-down"
            },
            "InstanceId": "i-09c124089c57fa7d0",
            "PreviousState": {
                "Code": 16,
                "Name": "running"
            }
        }
    ]
}
{
    "TerminatingInstances": [
        {
            "CurrentState": {
                "Code": 32,
                "Name": "shutting-down"
            },
            "InstanceId": "i-0a6eaa6a278c7ea37",
            "PreviousState": {
                "Code": 16,
                "Name": "running"
            }
        }
    ]
}
    All EC2 instances have been shutdown.
    Deleteing Volumes
    All Volumes are gone
    Releasing IP addresses...
    Public IPs are gone
    Deleting ELBs...
    ELBs are gone
    Deleting RDS instances...
    RDS instances are gone
    Deleting S3 buckets...
    S3 buckets gone
    Deleting security groups...

An error occurred (DependencyViolation) when calling the DeleteSecurityGroup operation: resource sg-060e4afc6df7eb4ba has a dependent object
xargs: aws: exited with status 255; aborting
```

**Comment by dmikusa-pivotal** [comment]
@jsdoeung-pivotal - the AWS API is a pain. The order in which you delete things matters heavily and when something doesn't get deleted in the right order you end up with this error. Unlike the other IaaS' AWS doesn't seem to have a recursive delete option, so there's no easy button.

Just have to kind of see what it's trying to delete, look at what is preventing that, and then adjust the script to delete things in the correct order. It infamously doesn't provide a lot of clues as to what is preventing the delete either.

If you still have this lab environment around there's two things you can do:

 1.) Easy way: go into the UI & try deleting the security groups. The AWS Console can often delete stuff recursively where the API cannot. It may let you delete the security group and then you can re-run the script and continue deleting the rest of stuff.

 2.) Hard way: go into the UI & try to see what is preventing it from deleting the security group. Look what resources are attached to the security group & what it's attached to. These associations are usually what holds up the delete. When you figure it out, take a look at the delete script and see if you can adjust the order to delete the dependency first. Then re-run the script. If all goes well, it'll delete everything else. Then submit a PR with what you changed.

**Comment by jsdoeung-pivotal** [comment]
@dmikusa-pivotal Yes, agree 💯  that the AWS API is a pain. I realized that there are things like IAM user groups, roles, policies and encryption keys that are created by terraform for a PCF install that we may also want to include in the cleaner script. I'll see what I can do and will submit a PR when ready. Thanks.
